stdlib/grpc/src/main/java/org/ballerinalang/net/grpc/builder/components/Method.java

```diff
@@ -93,9 +143,100 @@ public Method build() {
             String methodName = methodDescriptor.getName();
             String inputType = methodDescriptor.getInputType();
             inputType = inputType != null ? getMappingBalType(inputType) : null;
+            boolean primitiveInput = inputType != null && checkPrimitiveType(inputType);
             String outputType = methodDescriptor.getOutputType();
             outputType = outputType != null ? getMappingBalType(outputType) : null;
-            return new Method(methodName, methodId, inputType, outputType, methodType);
+            HttpRule httpExtension = methodDescriptor.getOptions().getExtension(AnnotationsProto.http);
+            String httpMethod = httpExtension.getPatternCase().name();
+            String httpPath = resolveHttpPath(httpExtension);
+            String httpBody = httpExtension.getBody();
+            List<Param> paramSet = getParamList(inputType, messageMap, httpExtension);
+            return new Method(methodName, methodId, inputType, primitiveInput, outputType, methodType, httpMethod,
+                    httpPath, httpBody, paramSet);
+        }
+
+        private String resolveHttpPath(HttpRule httpExtension) {
+            int number = httpExtension.getPatternCase().getNumber();
+            // the following http methods are mapped thee numbers
+            // 0:no extension / 2:get /3:put /4:post /5:delete /6:patch
+            switch (number) {
+                case 2:
+                    return httpExtension.getGet();
+                case 3:
+                    return httpExtension.getPut();
+                case 4:
+                    return httpExtension.getPost();
+                case 5:
+                    return httpExtension.getDelete();
+                case 6:
+                    return httpExtension.getPatch();
+                default:
+                    return "";
+            }
+        }
+
+        private List<Param> getParamList(String inputType, Map<String, Message> messageMap, HttpRule httpExtension) {
+            String httpMethod = httpExtension.getPatternCase().name();
+            if (httpMethod.equals(HTTP_PATTERN_NOT_SET) || inputType == null) {
+               return new ArrayList<>();
+            }
+            String httpPath = resolveHttpPath(httpExtension);
```

Comment:
Do we really need an allocated variable since this `httpPath` used just once?

Reply:
I removed the variable and directly applied the function at the one place it was used